### PR TITLE
kill chaos-tproxy while failing to apply config

### LINKED ISSUE #2660: bpm runs into deadlocked once errors occur in applying config of HTTPChaos
## Bug Report

**What version of Chaos Mesh are you using?**

v2.x.x

**What did you do?**

Encounter any errors in applying the config of HTTPChaos .

**What did you expect to see?**

HTTPChaos works after retrying or fixing config.

**What did you see instead?**

The `bpm` runs into deadlocked

**Reason**

The bpm will try to get an identifier lock when a subprocess starts (the instance is 0) and release the lock when the subprocess exits. Currently, chaos-tproxy won't exit when encountering any errors in applying config, and in this case, the PodHTTPChaos won't store the instance in status. 

**Solutions**
1. Kill the tproxy instance when encountering any errors in applying config
2. Store the instance id in the status of PodHTTPChaos once the tproxy starts.

### PR DESCRIPTION
**Pull request by Hexilee**
### What problem does this PR solve?

Close #2660

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [x] release-2.1
  - [x] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [x] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: pkg/chaosdaemon/httpchaos_server.go [hunk 1]
```diff
@@ -60,16 +60,24 @@ func (t stdioTransport) RoundTrip(req *http.Request) (resp *http.Response, err e
...
 
-func (s *DaemonServer) ApplyHttpChaos(ctx context.Context, in *pb.ApplyHttpChaosRequest) (*pb.ApplyHttpChaosResponse, error) {
+func (s *DaemonServer) ApplyHttpChaos(ctx context.Context, in *pb.ApplyHttpChaosRequest) (ret *pb.ApplyHttpChaosResponse, err error) {
 	log := log.WithValues("Request", in)
 	log.Info("applying http chaos")
 
 	if in.Instance == 0 {
-		if err := s.createHttpChaos(ctx, in); err != nil {
+		if err = s.createHttpChaos(ctx, in); err != nil {
 			return nil, err
```
Comment (Andrewmatilde) [inline pkg/chaosdaemon/httpchaos_server.go:69]: Add log error here or wrap err then return.

**Comment by Andrewmatilde** [comment]
Please add error logs or wrap err here,
```

	req, err := http.NewRequest(http.MethodPut, "/", bytes.NewReader(config))
	if err != nil {
		return nil, err
	}
	resp, err := transport.RoundTrip(req)
	if err != nil {
		return nil, err
	}
	log.Info("http chaos applied")
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}
```

**Comment by Hexilee** [comment]
> Please add error logs or wrap err here,
> 
> ```
> 
> 	req, err := http.NewRequest(http.MethodPut, "/", bytes.NewReader(config))
> 	if err != nil {
> 		return nil, err
> 	}
> 	resp, err := transport.RoundTrip(req)
> 	if err != nil {
> 		return nil, err
> 	}
> 	log.Info("http chaos applied")
> 	body, err := ioutil.ReadAll(resp.Body)
> 	if err != nil {
> 		return nil, err
> 	}
> ```

Wrapped

**Review by Andrewmatilde — APPROVED** [review]
LGTM

**Comment by Andrewmatilde** [comment]
PTAL @YangKeao , @STRRL

File: pkg/chaosdaemon/httpchaos_server.go [hunk 2]
```diff
@@ -100,19 +108,19 @@ func (s *DaemonServer) ApplyHttpChaos(ctx context.Context, in *pb.ApplyHttpChaos
 
 	req, err := http.NewRequest(http.MethodPut, "/", bytes.NewReader(config))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("fail to create request: %v", err)
```
Comment (STRRL) [inline pkg/chaosdaemon/httpchaos_server.go:111]: Please use `"github.com/pkg/errors"` `erorrs.Wrapf()` instead of `fmt.Errorf`

Comment (Hexilee) [reply pkg/chaosdaemon/httpchaos_server.go]: Fixed

File: pkg/chaosdaemon/httpchaos_server.go [hunk 3]
```diff
@@ -108,19 +116,19 @@ func (s *DaemonServer) applyHttpChaos(ctx context.Context, logger logr.Logger, i
 
 	req, err := http.NewRequest(http.MethodPut, "/", bytes.NewReader(config))
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "fail to create http request")
```
Comment (cwen0) [inline pkg/chaosdaemon/httpchaos_server.go:119]: Please refer this docs: https://github.com/YangKeao/rfcs/blob/refine-error-handling/text/2021-09-27-refine-error-handling.md#error-wrapping

Comment (cwen0) [reply pkg/chaosdaemon/httpchaos_server.go:119]: Not using `fail to` to describe this error.

Comment (Hexilee) [reply pkg/chaosdaemon/httpchaos_server.go]: Fixed

**Review by cwen0 — APPROVED** [review]
LGTM

**Pull request merged**